src/executables/master_heartbeat.c

```diff
@@ -4676,6 +4848,26 @@ hb_cluster_initialize (const char *nodes, const char *replicas)
       return ER_FAILED;
     }
 
+  if (hb_Cluster->num_ping_hosts == 0)
+    {
+      /* The ha_tcp_ping_hosts can be used instead of the ha_ping_hosts in case the ICMP protocol is disabled.
+         However, both system parameters cannot be set at the same time and the ha_tcp_ping_hosts works only when the ha_ping_hosts is not set. */
+      hb_Cluster->num_ping_hosts = hb_cluster_load_tcp_ping_host_list (prm_get_string_value (PRM_ID_HA_TCP_PING_HOSTS));
+      hb_Cluster->ping_timeout = prm_get_integer_value (PRM_ID_HA_PING_TIMEOUT);
+
+      if (hb_cluster_check_valid_ping_server () == false)
```

Comment:
Inside the `hb_cluster_check_valid_ping_server()`, 
`hb_check_ping ()` should be replaced with `hb_check_tcp_ping ()` if it is called by tcp ping context.

Reply:
Explain more for me. It already distinguish which version is called according to 'ping_host->port == 0'.

Reply:
Oh.. Sorry, I just grep with `hb_cluster_check_valid_ping_server` and didn't find out that function has been modified.